src/models/embeddingsList.ts

```diff
@@ -25,14 +25,14 @@ export class EmbeddingsList
   extends Array<Embedding>
   implements OpenAPIEmbeddingsList
 {
-  model: string;
+  model?: string;
   data: Array<Embedding>;
-  usage: EmbeddingsListUsage;
+  usage?: EmbeddingsListUsage;
 
   constructor(
-    model: string,
-    data: Array<Embedding>,
-    usage: EmbeddingsListUsage
+    model?: string,
+    data: Array<Embedding> = [],
+    usage?: EmbeddingsListUsage
```

Comment:
This change is related to this previous comment: https://github.com/pinecone-io/pinecone-ts-client/pull/240#discussion_r1683159720

Basically, I think what was happening is you had your custom class expecting non-optionals for all of these, but that doesn't align with the spec. There may have been cases where this was causing problems because of how this class is extending and implementing `Array` and the generated `EmbeddingsList`.

Reply:
sounds great !